features/steps/thens/existance.py

```diff
@@ -47,4 +47,26 @@ def step_impl(context, inst):
                         observed=observed_count,
                         expected=expected_count,
                         severity=OutcomeSeverity.ERROR
-                    )
\ No newline at end of file
+                    )
+
+def recursive_flatten(lst):
+    flattened_list = []
+    for item in lst:
+        if isinstance(item, (tuple, list)):
+            flattened_list.extend(recursive_flatten(item))
+        else:
+            flattened_list.append(item)
+    return flattened_list
+
+def get_previous_step_before_assertion(context):
+    for i, step in enumerate(context.scenario.steps):
+        if step.name == 'assert existence' and i > 0:
```

Comment:
Oh this is nice. Well done, just small comment that I would not hardcode the step name since anyway we have it.

```suggestion
        if step.name.lower() == context.step.name.lower() and i > 0:
```

Reply:
incorporated into `27347b`